openai_connector/models/openai_vision_session.py

```diff
@@ -0,0 +1,130 @@
+        selection=[],
+        required=True,
+        help="Defines the specific purpose of the OpenAI Vision session.",
+    )
+    llm_model = fields.Selection(
+        [("gpt-4o", "GPT-4o")],
+        string="Model",
+        default="gpt-4o",
+        required=True,
+    )
+    temperature = fields.Float(
+        default=0.7,
+        help="Sets response randomness (0–2)."
+        " Higher is more creative, lower is more focused.",
+    )
+    instruction = fields.Text(
+        help="System-level instruction for the assistant."
+        " Example: You are a helpful assistant that answers in Japanese."
+    )
+    previous_response_id = fields.Char(
+        string="Previous Response ID",
+        help="ID of previous response to continue chat (expires in 30 days).",
+    )
+    store_response = fields.Boolean(
+        help="If enabled, the API response is stored and the response ID is saved for"
+        " session continuity, allowing the conversation context to be preserved across"
+        " multiple calls."
+    )
+    web_search = fields.Boolean(
+        string="Use Web Search",
+        help="If enabled, the model can use web search to find information beyond "
+        "its training data.",
+    )
+    response_format_enabled = fields.Boolean(
+        string="Use Structured Output (JSON Schema)",
+        help="If enabled, the response will be structured according to "
+        "the Response Format Schema (JSON)",
+    )
+    response_format_schema = fields.Text(
+        string="Response Format Schema (JSON)",
+        default=lambda self: json.dumps(
+            {
+                "type": "object",
+                "properties": {"message": {"type": "string"}},
+                "required": ["message"],
+                "additionalProperties": False,
+                "strict": True,
+            },
+            indent=2,
+        ),
+    )
+
+    @api.constrains("response_format_schema")
+    def _constrains_response_format_schema(self):
+        for record in self:
+            try:
+                json.loads(record.response_format_schema or "{}")
+            except json.JSONDecodeError as e:
+                raise UserError(_("The JSON schema is invalid:\n{}").format(e)) from e
+
+    def _get_request_payload(self, input_datas):
+        self.ensure_one()
+        try:
+            json.loads(input_datas or "[]")
+        except json.JSONDecodeError as e:
+            raise UserError(_("The inputs are invalid: %s") % e) from e
+        request_payload = {
+            "model": self.llm_model,
+            "instructions": self.instruction or "",
+            "input": json.loads(input_datas) or [],
+            "temperature": self.temperature,
+            "store": self.store_response,
+            "tools": [{"type": "web_search_preview"}] if self.web_search else [],
+        }
+        if self.response_format_enabled:
+            schema = json.loads(self.response_format_schema or "{}")
```

Comment:
Just curious. What will be the response  if `response_format_schema` is `{}`. Will get empty response?